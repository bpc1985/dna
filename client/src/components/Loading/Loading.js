import React from "react";

import "./Loading.scss";

const Loading = () => (
  <div className="spinner">
    <div className="ldsDualRing" />
  </div>
);

export default Loading;
