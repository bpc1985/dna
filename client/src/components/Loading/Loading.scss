@import "stylesheets/index";

.spinner {
  display: flex;
  justify-content: center;
  align-items: center;
  margin-top: 6rem;

  .ldsDualRing {
    display: inline-block;
    width: 64px;
    height: 64px;
  }

  .ldsDualRing:after {
    content: ' ';
    display: block;
    width: 46px;
    height: 46px;
    margin: 1px;
    border-radius: 50%;
    border: 5px solid $pink;
    border-color: $pink transparent $pink transparent;
    animation: ldsDualRing 1.2s linear infinite;
  }

  @keyframes ldsDualRing {
    0% {
      transform: rotate(0deg);
    }
    100% {
      transform: rotate(360deg);
    }
  }
}