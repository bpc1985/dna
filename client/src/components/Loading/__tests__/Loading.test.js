import { render } from '@testing-library/react';
import React from "react";
import Loading from '../Loading';

describe('Loading', () => {
  it('should render component', () => {
    const { container } = render(
      <Loading />
    );
    expect(container).toMatchSnapshot();
  });
});
