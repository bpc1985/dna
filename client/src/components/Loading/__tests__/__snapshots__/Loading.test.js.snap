// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading should render component 1`] = `
<div>
  <div
    class="spinner"
  >
    <div
      class="ldsDualRing"
    />
  </div>
</div>
`;
